test/test_manual.jl

```diff
@@ -59,6 +60,116 @@ isdir(outdir) && rm(outdir, recursive=true)
     end
   end
 
+  @testset "ParallelTreeMesh" begin
+    @testset "partition!" begin
+      @testset "mpi_nranks() = 2" begin
+        mock((Trixi.mpi_nranks) => () -> 2) do mpi_ranks
```

Comment:
In general, if we keep `SimpleMock` in our tests, we should add at least a little bit of documentation on how it is supposed to be used (especiallly since the original documentation of `SimpleMock` is not very understandable itself IMHO). One approach could be to copy https://github.com/trixi-framework/Trixi.jl/blob/lbm3d/docs/src/testing.md and expand it with a section on "Mocking" or "Mock testing"?

Side note: I especially don't like that the first example in https://juliatesting.github.io/SimpleMock.jl/stable/#The-mock-Function does not work, while the second example takes forever and ever to compile...

Reply:
Actually, I found the documentation of `SimpleMock.jl` to be quite self-explanatory, but that's probably because I worked a lot with mocking frameworks in other languages.
But I agree, mocking with `SimpleMock.jl` takes way too much time for what it's supposed to do.

Reply:
I think the reason for it being slow is that it uses `Cassette`, which IIRC over-/rewrites the source code ad-hoc during compilation. That means that, e.g., if you are overwriting a common method such as `+`, it means *a lot* of recompilations, and it also means that time-to-first-solution is - as usual with Julia - not stellar. However, I have nothing against using `SimpleMock` in general, and our general rule is that we want to try new things in Trixi - thus why not. I'm just waiting to hear from @ranocha if he's ok with this and if yes, we can proceed here.

Reply:
Based on the other thread above, it looks like we're all not that satisfied with SimpleMock.jl....

Reply:
Resolving this as per https://github.com/trixi-framework/Trixi.jl/pull/361#discussion_r543968918